ui/components/component-library/text-field-search/text-field-search.js

```diff
@@ -40,15 +39,25 @@ TextFieldSearch.propTypes = {
    * The onChange handler of the TextFieldSearch
    */
   onChange: TextFieldBase.propTypes.onChange,
-  /**
-   * Show a clear button to clear the input
-   * Defaults to true
-   */
-  showClearButton: PropTypes.bool,
   /**
    * The onClick handler for the clear button
+   * Required unless showClearButton is false
+   *
+   * @param {object} props - The props passed to the component.
+   * @param {string} propName - The prop name in this case 'id'.
+   * @param {string} componentName - The name of the component.
    */
-  clearButtonOnClick: PropTypes.func,
+  clearButtonOnClick: (props, propName, componentName) => {
+    if (
+      props.showClearButton &&
+      (!props[propName] || !props.clearButtonProps?.onClick)
+    ) {
+      return new Error(
+        `${propName} is required unless showClearButton is false. Warning coming from ${componentName} ui/components/component-library/text-field-search/text-field-search.js`,
+      );
+    }
+    return null;
+  },
```

Comment:
So this custom propType makes the `clearButtonOnClick` required UNLESS `showClearButton` is set to `false`